### PR TITLE
[WIP]add istiod-central

### PR DESCRIPTION
**Pull request by irisdingbj**
Based on discussion from env WG yesterday:  https://docs.google.com/document/d/1EymnzMqa7hdmKn8sEoaf1elajyp4gXEDSUavxzXB3jw/edit#heading=h.3qdwucgp1qe8

### CODE REVIEW

**Comment by jwendell** [comment]
what is istiod-central?

File: Makefile.core.mk [hunk 1]
```diff
@@ -357,6 +357,25 @@ gen-check: check-no-modify gen check-clean-repo
 sync-configs-from-istiod:
 	cp manifests/charts/istio-control/istio-discovery/files/injection-template.yaml manifests/charts/istiod-remote/files/
 	cp manifests/charts/istio-control/istio-discovery/templates/istiod-injector-configmap.yaml manifests/charts/istiod-remote/templates/
+	cp manifests/charts/istio-control/istio-discovery/templates/configmap.yaml manifests/charts/istiod-remote/templates/
+	cp manifests/charts/istio-control/istio-discovery/templates/configmap-jwks.yaml manifests/charts/istiod-remote/templates/
+	cp manifests/charts/istio-control/istio-discovery/templates/telemetryv2_1.6.yaml manifests/charts/istiod-remote/templates/
+	cp manifests/charts/istio-control/istio-discovery/templates/telemetryv2_1.7.yaml manifests/charts/istiod-remote/templates/
```
Comment (linsun) [inline Makefile.core.mk:363]: @irisdingbj is this needed?  central isn't necessarily the control cluster.

Comment (linsun) [reply Makefile.core.mk:363]: nm, read it wrong.

**Comment by linsun** [comment]
@jwendell please check out this doc - https://docs.google.com/document/d/1EymnzMqa7hdmKn8sEoaf1elajyp4gXEDSUavxzXB3jw/edit#heading=h.3qdwucgp1qe8

@howardjohn FYI, a new component won't clutter logic to existing components.  still WIP, would appreciate your early feedback.

@irisdingbj thank you - this looks quite reasonable to me.

File: Makefile.core.mk [hunk 2]
```diff
@@ -357,6 +357,25 @@ gen-check: check-no-modify gen check-clean-repo
 sync-configs-from-istiod:
 	cp manifests/charts/istio-control/istio-discovery/files/injection-template.yaml manifests/charts/istiod-remote/files/
 	cp manifests/charts/istio-control/istio-discovery/templates/istiod-injector-configmap.yaml manifests/charts/istiod-remote/templates/
+	cp manifests/charts/istio-control/istio-discovery/templates/configmap.yaml manifests/charts/istiod-remote/templates/
+	cp manifests/charts/istio-control/istio-discovery/templates/configmap-jwks.yaml manifests/charts/istiod-remote/templates/
```
Comment (howardjohn) [inline Makefile.core.mk:361]: This is not needed

File: Makefile.core.mk [hunk 3]
```diff
@@ -357,6 +357,25 @@ gen-check: check-no-modify gen check-clean-repo
...
+	cp manifests/charts/istio-control/istio-discovery/templates/telemetryv2_1.6.yaml manifests/charts/istiod-remote/templates/
```
Comment (howardjohn) [inline Makefile.core.mk:362]: This will prevent users from having multiple remote clusters - only one should have the configs

**Review by howardjohn — CHANGES_REQUESTED** [review]
I strongly disagree with this approach. We can't keep duplicating everything over and over... Sorry if I suggested this was a good idea in the env meeting, I may have misunderstood the question

**Review by costinm — CHANGES_REQUESTED** [review]
Please don't merge, it's the wrong approach, back to spaghetti install.

Is there anything missing in the Istiod chart that would be needed in the 'central' istiod ? If yes, we can add it as an option. But I'm not aware of anything that would need to be configured and doesn't already have an option ( or 2 )

File: Makefile.core.mk (hunk 2 above)
Comment (linsun) [reply Makefile.core.mk:361]: I'm not sure about configmap-jwks as I haven't tried it... but we do find ConfigMap.yaml is required on remote data plane cluster else istio ingress gw won't reach running.

**Comment by linsun** [comment]
This is being reworked under a few smaller PRs to update existing components.

**Comment by irisdingbj** [comment]
as we decided to use flag other than introducing new component, so close this PR.

**Pull request closed without merging**